constantine/math_compiler/experimental/gpu_compiler.nim

```diff
@@ -54,25 +59,25 @@ type
     y*: DimWgsl
     z*: DimWgsl
 
-
 ## These are dummy elements to make CUDA block / thread index / dim
 ## access possible in the *typed* `cuda` macro. It cannot be `const`,
 ## because then the typed code would evaluate the values before we
 ## can work with it from the typed macro.
-let blockIdx* = NvBlockIdx()
-let blockDim* = NvBlockDim()
-let gridDim* = NvGridDim()
-let threadIdx* = NvThreadIdx()
+let blockIdx* {.builtin.} = NvBlockIdx()
+let blockDim* {.builtin.} = NvBlockDim()
+let gridDim* {.builtin.} = NvGridDim()
+let threadIdx* {.builtin.} = NvThreadIdx()
```

Comment:
maybe those backend specific builtins should be in a backend specific file

Reply:
Yeah, probably a good idea.